### PR TITLE
VXFM-4302 Inventory fails if MDM certificates not accepted

### PR DESCRIPTION
**Pull request by arunrordell**

### CODE REVIEW

File: lib/puppet/scaleio/transport.rb [hunk 1]
```diff
@@ -41,8 +41,14 @@ def get_scaleio_cookie
                            :accept => :json })
 
         rescue => ex
-          response = "NO MDM" if ex.class == RestClient::PreconditionRequired
-          Puppet.err "Failed to get cookie from ScaleIO Gateway with error %s" % [ex.message]
+          response_body = JSON.parse(ex.response.body)
```
Comment (gavin-scott) [inline lib/puppet/scaleio/transport.rb]: The asm-deployer code has some checks to see whether a body is returned, which makes me think we should maybe add those checks here also: https://github.com/dell-asm/asm-deployer/blob/529067ad4119fa4d7472d056280c392d01ef421e/lib/asm/provider/elementmanager/scaleio.rb#L687

File: lib/puppet/scaleio/transport.rb [hunk 2]
```diff
@@ -41,8 +41,14 @@ def get_scaleio_cookie
...
+
+          unless response_body["message"].include?("no MDM IP is set")
+            raise("%s and failed to get cookie" % [response_body["typeOfError"]]) if response_body["typeOfError"]
+            raise("Failed to get cookie from ScaleIO Gateway with error %s with httpStatusCode %s" % [response_body["message"], response_body["httpStatusCode"]])
```
Comment (gavin-scott) [inline lib/puppet/scaleio/transport.rb]: The above two lines make be a bit concerned we might lose some information from the exception that would not get logged. I would tend to do something like:

    raise("Failed to get cookie: %s: %s: %s" % [ex.class, ex.to_s, response_body])

File: lib/puppet/scaleio/transport.rb [hunk 3]
```diff
@@ -41,8 +41,14 @@ def get_scaleio_cookie
...
+          end
+          response = "NO MDM"
+          Puppet.err "Failed to get cookie from ScaleIO Gateway with error %s" % [response_body["message"]]
```
Comment (gavin-scott) [inline lib/puppet/scaleio/transport.rb]: Per above, account for the case of `response_body` being `nil`.

**Comment by arunrordell** [comment]
@gavin-scott I have updated this one, can you please review?

**Pull request merged**